src/lib.rs

```diff
@@ -162,3 +219,206 @@ impl LevelIndex {
         }
     }
 }
+
+bitflags::bitflags! {
+    #[repr(transparent)]
+    pub struct ChannelTypeQualifiers: u32 {
+        const LINEAR        = (1 << 0);
+        const EXPONENT      = (1 << 1);
+        const SIGNED        = (1 << 2);
+        const FLOAT         = (1 << 3);
+    }
+}
+
+bitflags::bitflags! {
+    #[derive(Default)]
+    #[repr(transparent)]
+    pub struct DataFormatFlags: u32 {
+        const STRAIGHT_ALPHA             = 0;
+        const ALPHA_PREMULTIPLIED        = (1 << 0);
+    }
+}
+
+// This can be obtained from std::mem::transmute::<f32, u32>(1.0f32) but we
+// want to support nostd. It is used for identifying normalized sample types
+// as in Unorm or Snorm
+const F32_1_AS_U32: u32 = 1065353216;
```

Comment:
Could [`core::mem::transmute`](https://doc.rust-lang.org/core/mem/fn.transmute.html) be used rather than hardcoding a magic number?

Reply:
This will be removed as noted below.